### ISSUE TITLE
Items Kept on Death - Certain untradables are still lost when not protected

### ISSUE THREAD

**Issue by Jademalo** [runelite/runelite#9167]
As mentioned in Discord, there are a few items that are lost on death if not protected, even though they are untradable.

The Wiki is slightly messy, and the information is split between the Items Kept on Death page, as well as the standard Death page. In addition, certain items have specific behaviour listed on their respective pages rather than in the master lists.

With the new Items Kept on Death plugin, these items are all shown as being kept on death in the same manner as any other untradable item.
However, in the event that these items are not one of the 3/4 that are protected, they have varying behaviour despite all being untradables.
For the items that are destroyed, they do not drop and will not be retrievable from item reclaimers such as Torfinn.

This information is all with respects to death outside of the Wilderness. 

**Always destroyed if not protected**
Ring of Charos+(a)
Lunar Staff 
Keris
Shadow Sword

**Always destroyed**
Clue Box
Looting Bag

**Always destroyed if not protected, even in safe locations**
Amulet of the Damned

**Dropped and Ornament kit lost if not protected, rather than being kept as an untradable**
(I believe this is all of the Bounty Hunter cosmetic rewards)
Dragon Pickaxe (or)
Volcanic Abyssal Whip
Frozen Abyssal Whip
Steam Battlestaff (or)
Lava Battlestaff (or)
Odium Ward (or)
Malediction Ward (or)
Granite Maul (or)

I am unsure about what happens when a Painted Dark Bow drops, since there is no information on the Wiki and I can't afford to test it. I believe it would function similarly to the other bounty hunter cosmetic upgrades, where it is dropped and the paint is lost.

Sources - 
https://oldschool.runescape.wiki/w/Items_Kept_on_Death#Items_that_are_never_kept
https://oldschool.runescape.wiki/w/Death
https://oldschool.runescape.wiki/w/Shadow_sword
https://oldschool.runescape.wiki/w/Lunar_staff
https://oldschool.runescape.wiki/w/Keris
https://oldschool.runescape.wiki/w/Amulet_of_the_damned
https://oldschool.runescape.wiki/w/Dragon_pickaxe_(or)
https://oldschool.runescape.wiki/w/Volcanic_abyssal_whip
https://oldschool.runescape.wiki/w/Frozen_abyssal_whip
https://oldschool.runescape.wiki/w/Steam_battlestaff_(or)
https://oldschool.runescape.wiki/w/Lava_battlestaff_(or)
https://oldschool.runescape.wiki/w/Odium_ward_(or)
https://oldschool.runescape.wiki/w/Malediction_ward_(or)
https://oldschool.runescape.wiki/w/Granite_maul_(or)
Hydrox tested the Ring of Charos (a), and confirmed it is destroyed.

**Comment by TheStonedTurtle** [comment]
> **Always destroyed**
Clue Box
Looting Bag

I believe these should already be handled correctly.

If i understand the rest of your post correctly the `Always destroyed if not protected` and `Dropped and ornate kit lost` items should act like their tradeable variant. I don't think the interface needs to explain that the ornate kits will be lost or the item will be destroyed, just that its in the right kept/lost location.

**Comment by Jademalo** [comment]
`Always destroyed if not protected` is destroyed entirely. It does not drop on the ground after a death if it is not one of the four protected items, and needs to be reacquired.

For the bounty hunter ornamented items you are correct, they essentially drop like their tradable variant while the ornament kit vanishes into thin air. 

I've rewritten this comment twice, and each time I've leaned closer towards actually being clearer and having the distinction of things that are totally unretrievable. To that end I disagree, and I think the interface should explain lost and destroyed.  
The whip cosmetics for example are worth 200k each, which for a lot of people isn't an insignificant sum, especially with regards to a death cost. At present there is absolutely no indication that if you die and the item drops that you will lose that. The assumption would be that it would drop on the ground to be picked up, just like any other item.
  
At the end of the day, the purpose of this interface is to know what will be lost if you die.  
The assumption is that if an item would be lost, it would be dropped on the ground and retrievable. My ticket highlights a few edge cases where this is not the case, and where the interface is ultimately still a bit unclear.

I would propose this -

- Three headings; Items kept on death, Items dropped on death, Items destroyed on death.
- Items that are unprotectable, such as the clue box and looting bag, are given a white outline, and the white outline text clarified to `Items with a white outline will always be destroyed`
- Items that will lose their ornament kit if dropped given a yelow outline, with the text `Items with a yellow outline will lose upgrades if dropped`

![image](https://user-images.githubusercontent.com/386846/59991025-7fa1ea80-963d-11e9-8604-50e17822dad9.png)

Apologies for the hacky image, but this is roughly what I mean.  
This has a clear distinction between all of the different edge cases. White line means the item is unprotectable, yellow line means ornament kit is lost. Ultimately, three things can happen - The item is kept, it's dropped, or it's destroyed.

**Comment by TheStonedTurtle** [comment]
I tested the granite maul (or) and it's kept on death even if not one of the 3 protected items, due to it being untradeable. The BH whips also specifically mention it has to be in the wilderness on the wiki.

As of now I'm going to assume all of those are functioning correctly in the interface minus the extra information about losing the kit in the wilderness.

The current issue is the below items should be treated as tradeable items but are not. (if they aren't protected by value they are lost)
```
Ring of Charos+(a)
Lunar Staff
Keris
Shadow Sword
Amulet of the Damned (degraded)
```
I believe the degraded amulet can be protected by value, if this is not the case please let me know (don't have an extra amulet/non ironman acc to test)

&nbsp;

As for your image about possible improvements, the different colored outlines are not easily do-able (will require an overlay) and editing the widget to add another header will be complicated and probably cause more issues. I think the easier solution will be to eventually update the plugin to outline all items with unique drop mechanics white and have hover text that explains whats happening to that specific item.

**Comment by Jademalo** [comment]
Alright, I guess the Wiki is incorrect. I'll try and sort out the wiki if I get the chance, thanks for testing that.

I can't test the Amulet myself, I don't have the funds to be able to test that. 
From the research I did, it seemed to imply that the amulet could be protected by value, but if it wasn't then it would be lost even in a PoH or something where everything should be kept. I'm not sure if the other 4 items behave the same way, or if the amulet is different. 

That seems like a fair compromise, obviously I'm unfamiliar with the complexities involved there. The main piece of information that I think would be unclear at a glance with that setup would be the differenece between the Looting Bag/Clue Box and the Charos etc, since the Charos etc can be protected whereas the bag and box cannot. Still, so long as the info is available somewhere and correct it should be fine.

**Comment by Adam-** [comment]
@TheStonedTurtle are these all fixed now?

**Comment by TheStonedTurtle** [comment]
The only change this issue required was adding the below untradeable items as Lost on Death if they are not protected by their Death Price. (Treated like tradeable items pretty much)
```
Amulet of the Damned (degraded)
Ring of Charos
Ring of Charos (a)
Lunar Staff
Keris (including poisoned variants)
Shadow Sword
```


The PR author also had some suggestions but based on the title this issue was fixed in #9161 via the `LostIfNotProtected.java` file

**Comment by Nightfirecat** [comment]
Closing based on 9161 addressing the lost items bug. If you feel strongly about the UI enhancements, please open a separate request for those. Thanks!
